src/disp_s1/solid_earth_tides.py

```diff
@@ -0,0 +1,247 @@
+from concurrent.futures import ThreadPoolExecutor
+from typing import Optional, Tuple
+
+import numpy as np
+import pandas as pd
+import rasterio
+from dolphin import Filename, io
+from opera_utils import get_zero_doppler_time
+from pysolid.solid import solid_grid
+from rasterio.coords import BoundingBox
+from rasterio.warp import transform_bounds
+from scipy.interpolate import RegularGridInterpolator
+
+from disp_s1._reference import ReferencePoint
+
+
+def load_raster_bounds_and_transform(
+    filename: Filename,
+) -> Tuple[BoundingBox, rasterio.crs.CRS, rasterio.Affine, Tuple[int, int]]:
+    """Load bounds, transform, CRS, and shape from a raster file."""
+    try:
+        with rasterio.open(filename) as src:
+            bounds = src.bounds
+            crs = src.crs
+            affine_transform = src.transform
+            height, width = src.shape
+        return bounds, crs, affine_transform, (height, width)
+    except Exception as e:
+        raise RuntimeError(f"Error loading raster file {filename}: {e}") from e
```

Comment:
```suggestion
    with rasterio.open(filename) as src:
        bounds = src.bounds
        crs = src.crs
        affine_transform = src.transform
        height, width = src.shape
    return bounds, crs, affine_transform, (height, width)
```
If the `.open` fails, rasterio will raise some exception, so we don't really gain much clarity by wrapping that in another `RuntimeError`